packages/openapi3/src/decorators.ts

```diff
@@ -32,3 +49,87 @@ export const $oneOf: OneOfDecorator = (
 export function getOneOf(program: Program, entity: Type): boolean {
   return program.stateMap(oneOfKey).get(entity);
 }
+
+const [getTagsMetadata, setTagsMetadata] = unsafe_useStateMap<
+  Type,
+  { [name: string]: OpenAPI3Tag }
+>(OpenAPI3Keys.tagsMetadata);
+
+/**
+ * Decorator to add metadata to a tag associated with a namespace.
+ * @param context - The decorator context.
+ * @param entity - The namespace entity to associate the tag with.
+ * @param name - The name of the tag.
+ * @param tagMetadata - Optional metadata for the tag.
+ */
+export const $tagMetadata: TagMetadataDecorator = (
```

Comment:
```suggestion
export const tagMetadataDecorator: TagMetadataDecorator = (
```

don't name those with $ for new ones